src/test/java/plugins/WorkflowPluginTest.java

```diff
@@ -231,7 +231,7 @@ private String createGitRepositoryInDockerContainer() throws IOException {
         GitContainer gitContainer = gitServer.get();
         try (GitRepo repo = new GitRepo()) {
             repo.changeAndCommitFoo("Initial commit");
-            repo.transferToDockerContainer(gitContainer.host(), gitContainer.port());
+            repo.transferToDockerContainer(GitContainer.addBracketsIfNeeded(gitContainer.host()), gitContainer.port());
```

Comment:
fix `GitRepo`?  after all you are passing in a host not a URL?

Reply:
This `transferToDockerContainer` method is using jsch which will create a session to connect to the host as an address, so requires brackets AFAIK

Reply:
> This `transferToDockerContainer` method is using jsch which will create a session to connect to the host as an address, so requires brackets AFAIK

to be clear host addresses do not contain square brackets...  Some usage of them in some formats (e.g. URLs) may require them to be escaped, but that is up to the implementation to do not a caller.
Today this is using `jsch` but tomorrow it could be using MINA or a different library which does not have this bug.

Reply:
I know, but today is today :-) and today based on the current `jsch` implementation this is not working under a dual-stack environment giving the following exception
```
Caused by: com.jcraft.jsch.JSchException: java.net.ConnectException: Connection refused
	at com.jcraft.jsch.Util.createSocket(Util.java:349)
	at com.jcraft.jsch.Session.connect(Session.java:215)
	at com.jcraft.jsch.Session.connect(Session.java:183)
	at org.jenkinsci.test.acceptance.plugins.git.GitRepo.transferToDockerContainer(GitRepo.java:289)
	... 42 more
```
So, this PR is trying to make it work there at the same time works under IPv4. Is there room for improvement? Sure thing, but at least this is better than nothing.

Reply:
I agree with James here, if jsch is not able to generate proper urls then the point to fix the issue is in the code that calls it and not before. A host is a host and a url is another thing, so your parameters should not be an improper host representation to hack how the url representation is generated by the jsch library...

Reply:
Addressed in https://github.com/jenkinsci/acceptance-test-harness/pull/1510/commits/5338f366038f3de2105851ac1ff39d83064583a8